### PR TITLE
docs: add realtime and segments

### PR DESCRIPTION
**Pull request by luorobin-a2z**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting
  - [ ] streaming ingestion
    - [ ] with Redshift Serverless
    - [ ] with provisioned Redshift

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: docs/en/analytics/realtime/index.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Realtime 
+Realtime lets you monitor clickstream data on your website or app in second-level latency. 
+
+## Prerequisites 
+1. You must enable realtime analysis for the data pipeline of your project.
+2. You must enable realtime streaming for the app.
+
+## Start and stop
+You can start and stop the realtime data streaming by switching on and off the toggle to avoid unnecessary cost. In the backend, this option starts and stops the Flink application which process the clickstream events.
```
Comment (jingnanl) [inline docs/en/analytics/realtime/index.md:9]: process -> processes

File: docs/en/analytics/segment/index.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+To access Segments, follow below steps:
+
+1. Go to **Clickstream Analytics on AWS Console**, in the **Navigation Bar**, click on "**Analytics Studio**", a new tab will be opened in your browser.
+2. In the Analytics Studio page, click the **Segments** icon in the left navigation panel.
+
+## Create a segment
+1. Click on **Create Segment +** button
+2. Input **Segment name** and **Description**
+3. Specify a **Refresh method**
+    - Manual - Segment will be refreshed when user manually trigger refresh
```
Comment (jingnanl) [inline docs/en/analytics/segment/index.md:14]: trigger -> triggers

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @luorobin-a2z on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1533)
###### _updated: 7/12/2024, 10:07:18 (UTC+0)_

**Comment by zxkane** [comment]
# SonarQube Code Analytics
## Quality Gate passed

passed

## Additional information
*The following metrics might not affect the Quality Gate status but improving them will improve your project code quality.*

## Issues
bug   [0 Bugs](http://sonarqube:9000/project/issues?id=pr-1533&resolved=false&sinceLeakPeriod=true&types=BUG)

vulnerability   [0 Vulnerabilities](http://sonarqube:9000/project/issues?id=pr-1533&resolved=false&sinceLeakPeriod=true&types=VULNERABILITY)

code_smell   [0 Code Smells](http://sonarqube:9000/project/issues?id=pr-1533&resolved=false&sinceLeakPeriod=true&types=CODE_SMELL)

## Coverage and Duplications
*No data* **Coverage**

*No data* **Duplication**

File: docs/en/analytics/segment/index.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    - Automatically - Segment will be refreshed periodically. You can configure interval (e.g., Daily, Weekly, Monthly) and timing (e.g., 8AM).
```
Comment (jingnanl) [inline docs/en/analytics/segment/index.md:15]: Automatically -> Automatic
Also support custom cron or rate expression

File: docs/en/analytics/segment/index.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+1. Go to **Clickstream Analytics on AWS Console**, in the **Navigation Bar**, click on "**Analytics Studio**", a new tab will be opened in your browser.
+2. In the Analytics Studio page, click the **Segments** icon in the left navigation panel.
+
+## Create a segment
+1. Click on **Create Segment +** button
+2. Input **Segment name** and **Description**
+3. Specify a **Refresh method**
+    - Manual - Segment will be refreshed when user manually trigger refresh
+    - Automatically - Segment will be refreshed periodically. You can configure interval (e.g., Daily, Weekly, Monthly) and timing (e.g., 8AM). 
+4. Specify a **Expirations settings**. If you specified Automatically as refresh method, you will need to set an expiration date for a segment after which the segment will stop auto refresh.
```
Comment (jingnanl) [inline docs/en/analytics/segment/index.md:16]: Automatically -> Automatic

Specify an Expiration settings.

Currently the expiration settings is not mandatory.

File: docs/en/analytics/segment/index.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+|**User has done in sequence**| Specify an event behavior sequence. Users will only be selected if they trigger these events in the specified order.| <li>Within a session: user must done the selected events within a session </li> <li>Across sessions: user has done the selected events in one or multiple sessions</li> <li>Directly followed: The next event needs to be immediately after the previous one</li> <li> Indirectly followed: There can be other events between the specified step sequence</li>||
+|**User is**/**User is not**|Select users by user attributes.| User attributes|<li>is null</li><li>is not null</li><li>=</li><li>!=</li> <li>in</li><li>not in</li><li>contains</li><li>not contains</li> |
+|**User in segment/User in not segment**| Select users based on whether user belong to a segment or not.| Segments||
```
Comment (jingnanl) [inline docs/en/analytics/segment/index.md:38]: User not in segment

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @luorobin-a2z on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1533)
###### _updated: 7/15/2024, 03:52:37 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @luorobin-a2z on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.70|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1533)
###### _updated: 7/15/2024, 03:52:57 (UTC+0)_

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @zxkane on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1533)
###### _updated: 7/15/2024, 15:36:49 (UTC+0)_

**Review by zxkane — CHANGES_REQUESTED** [review]
- [x] WARNING -  Doc file 'analytics/index.md' contains a relative link './segments/index.md', but the target 'analytics/segments/index.md' is not found among documentation files.

File: docs/en/analytics/index.md [hunk 6]
```diff
@@ -5,8 +5,10 @@ Analytics Studio is a unified web interface for business analysts or data analys
 Below are the modules included in the Analytics Studio:
 
 - [Dashboard](./dashboard/index.md). View out-of-the-box dashboard and custom dashboards.
+- [Realtime](./realtime/index.md). View realtime dashboard.
 - [Explorations](./explore/index.md). Advance analytics models for user to uncover insights beyond the out-of-the-box dashboard.
 - [Analyses](./analyzes/index.md). Author interface for creating and modifying dashboard, as well as managing datasets.
+- [Segments](./segments/index.md). Create user segments.
```
Comment (zxkane) [inline docs/en/analytics/index.md:11]: `./segment/index.md`

**Review by zxkane — CHANGES_REQUESTED** [review]
- [ ] miss the steps to enable/disable the stream at application level

**Comment by zxkane** [comment]
### SonarQube Quality Gate Result
- **Result**: :white_check_mark: OK
- Triggered by @luorobin-a2z on `pull_request`

| Metric | Status | Value | Error Threshold |
|:------:|:------:|:-----:|:---------------:|
|Reliability rating|:white_check_mark: OK|1|> 1|
|Security rating|:white_check_mark: OK|1|> 1|
|Sqale rating|:white_check_mark: OK|1|> 1|
|Coverage|:white_check_mark: OK|83.80|< 80|
|Duplicated lines density|:white_check_mark: OK|8.60|> 30|
|Blocker violations|:white_check_mark: OK|0|> 0|
|Bugs|:white_check_mark: OK|0|> 0|
|Code smells|:white_check_mark: OK|9|> 40|
|Critical violations|:white_check_mark: OK|0|> 0|
|Major violations|:white_check_mark: OK|0|> 0|
|Vulnerabilities|:white_check_mark: OK|0|> 0|

[View on SonarQube](http://sonarqube:9000/dashboard?id=pr-1533)
###### _updated: 7/17/2024, 09:01:04 (UTC+0)_

**Review by zxkane — APPROVED** [review]

**Pull request merged**